supabase/functions/postmark/addNoteToContact.ts

```diff
@@ -0,0 +1,123 @@
+import { supabaseAdmin } from '../_shared/supabaseAdmin.ts';
+
+export const addNoteToContact = async ({
+    salesEmail,
+    email,
+    domain,
+    firstName,
+    lastName,
+    noteContent,
+}: {
+    salesEmail: string;
+    email: string;
+    domain: string;
+    firstName: string;
+    lastName: string;
+    noteContent: string;
+}) => {
+    const { data: sales, error: fetchSalesError } = await supabaseAdmin
+        .from('sales')
+        .select('*')
+        .eq('email', salesEmail)
+        .maybeSingle();
+    if (fetchSalesError)
+        return new Response(
+            `Could not fetch sales from database, email: ${salesEmail}`,
+            { status: 500 }
+        );
+    if (!sales)
+        return new Response(
+            `Unable to find sales in database, email: ${salesEmail}`,
+            { status: 403 }
```

Comment:
**nit-pick:**
```suggestion
            { status: 404 }
```

Reply:
403 is a Postmark special status code, but I'll add a comment to explain it